File: src/api/writeFile.ts
import { writeBinaryFile } from '@tauri-apps/api/fs'
import { FileEncodings } from '~/types'

import invoke from './invoke'

writeBinaryFile

const writeFile = async (path: string, contents: string, encoding: FileEncodings): Promise<void> => {
  await invoke('write_file', { path, contents, encoding })
}

export default writeFile

File: src-tauri/src/commands/write_file.rs
use std::fs;

use crate::errors::Error;

#[tauri::command]
pub async fn write_file(path: &str, contents: &str, encoding: &str) -> Result<(), Error> {
  match encoding {
    "utf8" => {
      let buffer = contents.as_bytes().to_vec();

      fs::write
        (path, buffer)
        .map_err(Into::into)
        .map(|_| ())
    },
    _ => {
      let buffer = contents.as_bytes().to_vec();

      fs::write
        (path, buffer)
        .map_err(Into::into)
        .map(|_| ())
    }
  }
}
